spec/models/post_spec.rb

```diff
@@ -35,5 +35,16 @@
         expect(post.up_votes).to eq(1)
       end
     end
+
+    describe "#save_with_initial_vote" do
+      it "saves a post only if the initial vote saves as well" do
+        post = described_class.new(title: "Check One Two",
+                        body:  "Sibilance. Sibilance.",
+                        user:  authenticated_user,
+                        created_at: Time.new)
```

Comment:
Ah you are right! Awesome catch!! :trophy: :+1: :smiley:  

I'm not sure why I was under the impression that the method was calling `save` last. In this case, I would just get rid of that test. Mostly because we are already checking the `Vote` association in the following specs. I would do something like:

```ruby
let(:post) { associated_post_unsaved } # Or some other factory method (e.g. Fabrication gem)
...
describe "#save_with_initial_vote" do
  it "saves a post only if the initial vote saves as well" do
    post.save_with_initial_vote
    expect(post).to be_persisted
  end

  it "associates a Vote object with the post" do
    post.save_with_initial_vote
    expect(post.vote).to be_a Vote # or .to_not be_nil
    expect(post.vote).to be_persisted
  end
end
```

I also like how the second set of tests seems a bit more readable.